TileFusionManager.java

```diff
@@ -0,0 +1,247 @@
+            for (int row = 1; row < gridSize; row++) {
+                if (gameBoard[row][col] != 0) {
+                    int newRow = row;
+                    while (newRow > 0 && gameBoard[newRow - 1][col] == 0) {
+                        gameBoard[newRow - 1][col] = gameBoard[newRow][col];
+                        gameBoard[newRow][col] = 0;
+                        newRow--;
+                        hasMoved = true;
+                    }
+                    if (newRow > 0 && gameBoard[newRow - 1][col] == gameBoard[newRow][col]) {
+                        gameBoard[newRow - 1][col] *= 2;
+                        currentScore += gameBoard[newRow - 1][col];
+                        gameBoard[newRow][col] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
+      //Moves all tiles down and merges adjacent tiles with the same value.
+    private void moveDown() {
+        for (int col = 0; col < gridSize; col++) {
+            for (int row = gridSize - 2; row >= 0; row--) {
+                if (gameBoard[row][col] != 0) {
+                    int newRow = row;
+                    while (newRow < gridSize - 1 && gameBoard[newRow + 1][col] == 0) {
+                        gameBoard[newRow + 1][col] = gameBoard[newRow][col];
+                        gameBoard[newRow][col] = 0;
+                        newRow++;
+                        hasMoved = true;
+                    }
+                    if (newRow < gridSize - 1 && gameBoard[newRow + 1][col] == gameBoard[newRow][col]) {
+                        gameBoard[newRow + 1][col] *= 2;
+                        currentScore += gameBoard[newRow + 1][col];
+                        gameBoard[newRow][col] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
+     // Moves all tiles left and merges adjacent tiles with the same value.
+    private void moveLeft() {
+        for (int row = 0; row < gridSize; row++) {
+            for (int col = 1; col < gridSize; col++) {
+                if (gameBoard[row][col] != 0) {
+                    int newCol = col;
+                    while (newCol > 0 && gameBoard[row][newCol - 1] == 0) {
+                        gameBoard[row][newCol - 1] = gameBoard[row][newCol];
+                        gameBoard[row][newCol] = 0;
+                        newCol--;
+                        hasMoved = true;
+                    }
+                    if (newCol > 0 && gameBoard[row][newCol - 1] == gameBoard[row][newCol]) {
+                        gameBoard[row][newCol - 1] *= 2;
+                        currentScore += gameBoard[row][newCol - 1];
+                        gameBoard[row][newCol] = 0;
+                        hasMoved = true;
+                    }
+                }
+            }
+        }
+    }
+     // Moves all tiles right and merges adjacent tiles with the same value.
+    private void moveRight() {

```

Comment:
There are lots of things common in the up down right and left method! Can you optimize the code ?